### PR TITLE
Fixed status code 1 issue with backward compatibility

### PR DESCRIPTION
**Pull request by samyakOO7**
Fixed status code 1 issue while running backward compatibility check when no spec files change is found

### CODE REVIEW

File: application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt [hunk 1]
```diff
@@ -92,7 +92,10 @@ abstract class BackwardCompatibilityCheckBaseCommand : Callable<Unit> {
         ).filter {
             File(it).exists() && File(it).isValidSpec()
         }.toSet().also {
-            if(it.isEmpty()) exitWithMessage("No specs were changed, skipping the check.")
+            it.takeIf { it.isEmpty() }?.run {
```
Comment (harikrishnan83) [inline application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt]: Just curious, why do we have to switch to `takeIf`? Can we not make the `if(it.isEmpty())` condition body a block body and just put the logger line and exitProcess with 0?

**Review by harikrishnan83 — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you please look into my comment?

File: application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt (hunk 1 above)
Comment (samyakOO7) [reply application/src/main/kotlin/application/backwardCompatibility/BackwardCompatibilityCheckBaseCommand.kt]: Yes, just from a functional programming perspective I added but it's unclear and extra checks not needed so rectified it

**Review by harikrishnan83 — APPROVED** [review]
Looks good to me

**Pull request merged**